### DISCUSSION TITLE
I need help with getStaticProps

### DISCUSSION THREAD

**Discussion by ConstructionIsComplite** [vercel/next.js#55873] [Help] [answered]
### Summary

My site page is located inside the app directory and I used generation without getStaticProps, with my own async function, and everything worked for me, then I found out that it is possible to use getStaticProps better. And I did something like in this tutorial: https://nextjs.org/learn/basics/data-fetching/with-data, but the data was undefined. I realized that getStaticProps cannot be used in the app directory, so I created a pages folder and put the folder with the page there, but then Next.js cannot find the path to it and a 404 error appears.

What am I doing wrong?

**Comment by Kfnig24** [comment, marked as answer]
I think Nextjs doesn't allow to use both app and page directory, and if you want to switch, it may need some configuration.

**Reply by ConstructionIsComplite** [reply]
So can i use only page directory, without app directory?

**Reply by Kfnig24** [reply]
I think but you'll probably need to do some configuration (I don't know which, you'll have to search by yourself)
